### PR TITLE
Battle doesn't stop, after AI kills the last creature

### PR DESCRIPTION
**Pull request by shprotru**
Fix for #1948

### CODE REVIEW

File: src/fheroes2/battle/battle_arena.cpp [hunk 1]
```diff
@@ -368,6 +368,10 @@ void Battle::Arena::TurnTroop( Unit * current_troop )
             // apply action
             ApplyAction( actions.front() );
             actions.pop_front();
+            if ( !army1->isValid() || !army2->isValid() ) {
```
Comment (ihhub) [inline src/fheroes2/battle/battle_arena.cpp:371]: I think `if ( !BattleValid() )` does exactly the same. We need to just add a break there. Could you please test such change on the given corner case and for few normal battles?

File: src/fheroes2/battle/battle_arena.cpp [hunk 2]
```diff
@@ -369,14 +369,16 @@ void Battle::Arena::TurnTroop( Unit * current_troop )
...
             actions.pop_front();
 
+            // check end battle
+            if ( !BattleValid() ) {
+                end_turn = true;
+                break;
+            }
+
             // rescan orders
             if ( armies_order )
```
Comment (ihhub) [inline src/fheroes2/battle/battle_arena.cpp:379]: Hi @shprotru , could you please test the same situation but with activated window for army order (just hardcode values for testing) and check how this window shows for provided save? I think we should have just added break to the existing if-case without moving the code above.

File: src/fheroes2/battle/battle_arena.cpp [hunk 3]
```diff
@@ -374,12 +374,14 @@ void Battle::Arena::TurnTroop( Unit * current_troop )
...
+            if ( !BattleValid() ) {
                 end_turn = true;
+                break;
+            }
 
             // good morale
-            if ( !end_turn && current_troop->isValid() && !current_troop->Modes( TR_SKIPMOVE ) && current_troop->Modes( TR_MOVED ) && current_troop->Modes( MORALE_GOOD )
-                 && BattleValid() ) {
+            if ( !end_turn && current_troop->isValid() && !current_troop->Modes( TR_SKIPMOVE ) && current_troop->Modes( TR_MOVED )
+                 && current_troop->Modes( MORALE_GOOD ) ) {
```
Comment (ihhub) [inline src/fheroes2/battle/battle_arena.cpp:384]: Nice catch!

**Comment by ihhub** [comment]
@shprotru thank you very much for fixing this issue!
